src/runtime_src/core/edge/sw_emu/generic_pcie_hal2/shim.cxx

```diff
@@ -541,34 +541,22 @@ namespace xclswemuhal2 {
           xilinxInstall = ".";
         }
 
-        std::string modelDirectory("");
-        if (boost::filesystem::exists(xilinxInstall + "/data/emulation/unified/sw_emu/zynqu/model/genericpciemodel"))
-          modelDirectory = xilinxInstall + "/data/emulation/unified/sw_emu/zynqu/model/genericpciemodel";
-        else
-          modelDirectory = xilinxInstall + "/data/emulation/unified/cpu_em/zynqu/model/genericpciemodel";
-
-#if defined(__aarch64__)
-        if (boost::filesystem::exists(xilinxInstall + "/data/emulation/unified/sw_emu/zynqu/model/genericpciemodel"))
-          modelDirectory = xilinxInstall + "/data/emulation/unified/sw_emu/zynqu/model/genericpciemodel";
-        else
-          modelDirectory = xilinxInstall + "/data/emulation/unified/cpu_em/zynqu/model/genericpciemodel";
-#elif defined(__arm__)
-        if (boost::filesystem::exists(xilinxInstall + "/data/emulation/unified/sw_emu/zynq/model/genericpciemodel"))
-          modelDirectory = xilinxInstall + "/data/emulation/unified/sw_emu/zynq/model/genericpciemodel";
-        else
-          modelDirectory = xilinxInstall + "/data/emulation/unified/cpu_em/zynq/model/genericpciemodel";
-#endif
+        std::string modelDirectory = xilinxInstall + "/data/emulation/unified/sw_emu/generic_pcie/model/genericpciemodel";
 
         FILE *filep;
         if ((filep = fopen(modelDirectory.c_str(), "r")) != nullptr)
         {
           // file exists
+          if (access(modelDirectory.c_str(), X_OK) != 0)
+          {
+            std::cout << "genericpciemodel binary does not have executable permission." << std::endl;
```

Comment:
If executable permissions are not present then don't you think it needs to exit here?

Reply:
Yes, please check, updated as per suggestion.

Reply:
checked.